### DISCUSSION TITLE
High disk usage

### DISCUSSION THREAD

**Discussion by shm-dmitry** [crate/crate#10380] [General] [answered]
Hi.
Sometimes I see large IOPS on one of nodes.
![Screenshot_20200817_102040](https://user-images.githubusercontent.com/19342331/90368789-50d1a980-e073-11ea-8009-a3297c32b23e.png)

I disabled automatic ANALYZE execution (I set `stats.service.interval = 0`), but sometimes I see this problem.
How can I found query, produces this IOPS?

**Comment by mfussenegger** [comment]
There currently is no easy way to correlate disk IO to specific queries.

It's also not certain that this is caused by queries at all, it could be caused by other tasks like rebalance operations which result in replication activity.

Or it could be caused by using the SNAPSHOT feature and the spike correlates to the backups you take.


If it is caused by a query it is typically a query that operates on a large amount of data. So either queries without `WHERE` clause, or where it matches most of the data.

Another common case of high read disk IO is if you're using predicates that cannot utilize the index. For example if you wrap a scalar around a column - something like `WHERE lower(x) = 'foo'`. This operates like a table scan - it reads the value for `x` for each row to do the comparison.

We typically recommend to set up monitoring with all kinds of JMX metrics that CrateDB expose. Based on that it is often possible to see some kind of relation between different metrics that give a better idea what is going on.

**Comment by shm-dmitry** [comment, marked as answer]
I found problem in my code.
I disabled default statistics collection (because it starts by day but I want to start it by night).
I created my custom job executes `ANALYZE` via jdbc at one in the morning.
And after it I saw large IOPS in the afternoon.
I disabled my job yesterday, It was help to me (at least at this moment).
Problem was in job - it run `ANALYZE` and then `OPTIMIZE` on all tables.. But `OPTIMIZE blob.xxx` was failed, so my job restarts and restarts and restarts....

I will read later about JMX metrics - maybe it will help me to optimize some tables, thank you.
